smrf/distribute/precipitation.py

```diff
@@ -282,6 +284,12 @@ def distribute(self, data, dpt, precip_temp, ta, time, wind, temp,
                         temp,
                         self.config,
                         self.metadata)
+                # Scale input data based on config
+                if input_scalar_key is not None:
```

Comment:
Good catch! My thought process was that we wanted to perform undercatch correction before scaling, but looking back at the undercatch correction methods, the order won't make a difference. This was an oversight on my part.

Reply:
If this is gridded data, SMRF should not be doing undercatch correction since it's not measured data. For measured data, it shouldn't matter since you could apply the factor outside SMRF (to the csv files) as well.

Reply:
This should be addressed by the latest round of commits.